controllers/etcd_controller.go

```diff
@@ -1045,3 +1055,24 @@ func (r *EtcdReconciler) claimConfigMaps(etcd *druidv1alpha1.Etcd, selector labe
 	cm := NewEtcdDruidRefManager(r, etcd, selector, etcdGVK, canAdoptFunc)
 	return cm.ClaimConfigMaps(ss)
 }
+
+// SetupWithManager sets up manager with a new controller and r as the reconcile.Reconciler
+func (r *EtcdReconciler) SetupWithManager(mgr ctrl.Manager, workers int, ignoreOperationAnnotation bool) error {
+	predicates := []predicate.Predicate{
+		druidpredicates.GenerationChangedPredicate{},
+		druidpredicates.LastOperationNotSuccessful(),
```

Comment:
What about shoots with `lastOperation: Failed`? In the extensions we stop reconciling related resources.
https://github.com/gardener/gardener-extensions/blob/master/pkg/controller/infrastructure/controller.go#L69

/cc @rfranzke

Reply:
Nice catch, thanks. Though, it's not working for the regular extensions right now anyways, so I suggest to exclude it for now and fix it afterwards in one go for all components.

Reply:
Okay true. Does it at make sense (also for the extensions) to check the failed state at the beginning of a reconciliation and if it's failed we return w/o re-queuing?

Reply:
Yep, sounds good :)

Reply:
@timuthy We decided to postpone this for the next release - the druid is not yet doing anything with the `extensions.gardener.cloud/v1alpha1.Cluster` resource - but this is where the information where the shoot's last operation is `Failed`. As it doesn't work for the regular extension controllers yet let's proceed now without this check and then fix it altogether in the extensions + druid.